CMakeLists.txt

```diff
@@ -798,7 +798,7 @@ set (Seastar_PRIVATE_CXX_FLAGS
   -Wno-error=deprecated)
 
 include (CheckGcc107852)
-if (NOT Cxx_Compiler_BZ107852_Free)
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT Cxx_Compiler_BZ107852_Free)
```

Comment:
probably a better fix is to guard the check with `CMAKE_CXX_COMPILER_ID STREQUAL "GNU"` , like
```cmake
if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
  include (CheckGcc107852)
  if (NOT Cxx_Compiler_BZ107852_Free)
    list (APPEND Seastar_PRIVATE_CXX_FLAGS
     -Wno-error=stringop-overflow
     -Wno-error=array-bounds)
  endif ()
endif ()
```

Reply:
ok